pkg/activator/net/throttler.go

```diff
@@ -108,6 +109,34 @@ func newRevisionThrottler(revID types.NamespacedName,
 	}
 }
 
+// pickP2C implements power of two choices algorithm for fast load balancing.
+// See here: https://www.eecs.harvard.edu/~michaelm/postscripts/mythesis.pdf.
+// The function picks a target to send request to, given list of `podIPTracker` objects, a
+// callback to release the slot is returned as well.
+func pickP2C(tgts []*podIPTracker) (string, func()) {
+	if len(tgts) == 0 {
```

Comment:
We don't want to return `tgrs[0]` in case of 1.
Since now we have 1 pod, in a second we have 2. And if we don't update the depth on this one, we will make bad routing decisions.

Reply:
true wrt. the point on the `1` case :(.

Reply:
OK< redid in a good way.